mina-core/src/main/java/org/apache/mina/filter/executor/OrderedThreadPoolExecutor.java

```diff
@@ -197,8 +197,8 @@ public OrderedThreadPoolExecutor(int corePoolSize, int maximumPoolSize, long kee
         }
 
         // Now, we can setup the pool sizes
-        super.setCorePoolSize(corePoolSize);
         super.setMaximumPoolSize(maximumPoolSize);
+        super.setCorePoolSize(corePoolSize);
```

Comment:
Base on the code here, it does not seems like this will do any differences to different JDK version. Even if it does this would not be the part of the issue that mina could fix?